### PR TITLE
A set should perform better for search/remove operations

### PR DESCRIPTION
**Pull request by f3ath**

### CODE REVIEW

File: lib/src/injector_base.dart [hunk 1]
```diff
@@ -97,11 +97,7 @@ class Injector {
     final factory = _factoryMap[identity];
     final factoryId = factory.hashCode;
 
-    if (_factoryCallIds.contains(factoryId)) {
-      throw CircularDependencyException(type: T.toString());
-    }
-
-    _factoryCallIds.add(factoryId);
+    _factoryCallIds.add(factoryId) || (throw CircularDependencyException(type: T.toString()));
```
Comment (JulianBissekkou) [inline lib/src/injector_base.dart:100]: sneaky that you used the `||`  as a control flow statement 🙈 
I would still prefer an if here for readability 👍

**Comment by JulianBissekkou** [comment]
Thanks for submitting this PR 👍

File: lib/src/injector_base.dart (hunk 1 above)
Comment (f3ath) [reply lib/src/injector_base.dart:100]: I did that to avoid comparing with false. Would you prefer `!` or ` == false`? IMO both are rather ugly.

Comment (JulianBissekkou) [reply lib/src/injector_base.dart:100]: I prefer `!` 🙈

**Pull request merged**